### PR TITLE
Add some basic model classes for new auth lib

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AuthHelper.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import com.microsoft.azure.tools.auth.model.AuthType;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+public class AuthHelper {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AuthHelper.java:20]: Most code comes from https://github.com/microsoft/azure-maven-plugins/blob/52071f368ef064ae65c7e8bb00b99815ba428544/azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+package com.microsoft.azure.tools.auth.model;
+
+/**
+ * The auth type which user may define in configuration.
+ */
+public enum AuthType {
+    AUTO,
+    AZURE_CLI,
+    MANAGED_IDENTITY,
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java:15]: `MANAGED_IDENTITY` refers to Service principal?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthMethod.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    CLOUD_SHELL,
+    OAUTH2,
+    DEVICE_CODE,
+    VISUAL_STUDIO,
+    VSCODE,
+    INTELLIJ_IDEA,
+    MANAGED_IDENTITY,
+    MAVEN_CONFIGURATION,
+    MAVEN_SETTINGS
+}
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthMethod.java:23]: New line in file end

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AuthHelper.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                return AuthType.DEVICE_CODE;
+            case "managed_identity":
+                return AuthType.MANAGED_IDENTITY;
+            case "oauth2":
+                return AuthType.OAUTH2;
+            case "visual_studio":
+                return AuthType.VISUAL_STUDIO;
+            case "auto":
+                return AuthType.AUTO;
+        }
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AuthHelper.java:73]: how about move these witch into AuthType as a method such as parseType(String) ?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java (hunk 2 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java:19]: No, it is a new auth in azure resources

Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java:19]: image

**Review by Flanker32 — APPROVED** [review]

**Pull request merged**